src/pages/ValidateLoginPage/index.website.tsx

```diff
@@ -10,51 +8,16 @@ import Navigation from '@libs/Navigation/Navigation';
 import * as Session from '@userActions/Session';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps as validateLinkDefaultProps, propTypes as validateLinkPropTypes} from './validateLinkPropTypes';
+import type {ValidateLoginPageOnyxProps, ValidateLoginPageProps} from './types';
 
-const propTypes = {
-    /** The accountID and validateCode are passed via the URL */
-    route: validateLinkPropTypes,
-
-    /** Session of currently logged in user */
-    session: PropTypes.shape({
-        /** Currently logged in user authToken */
-        authToken: PropTypes.string,
-    }),
-
-    /** The credentials of the person logging in */
-    credentials: PropTypes.shape({
-        /** The email the user logged in with */
-        login: PropTypes.string,
-
-        /** The validate code */
-        validateCode: PropTypes.string,
-    }),
-
-    /** The details about the account that the user is signing in with */
-    account: PropTypes.shape({
-        /** Whether a sign on form is loading (being submitted) */
-        isLoading: PropTypes.bool,
-    }),
-};
-
-const defaultProps = {
-    route: validateLinkDefaultProps,
-    session: {
-        authToken: null,
-    },
-    credentials: {},
-    account: {},
-};
-
-function ValidateLoginPage(props) {
-    const login = lodashGet(props, 'credentials.login', null);
-    const autoAuthState = lodashGet(props, 'session.autoAuthState', CONST.AUTO_AUTH_STATE.NOT_STARTED);
-    const accountID = lodashGet(props.route.params, 'accountID', '');
-    const validateCode = lodashGet(props.route.params, 'validateCode', '');
-    const isSignedIn = Boolean(lodashGet(props, 'session.authToken', null));
-    const is2FARequired = lodashGet(props, 'account.requiresTwoFactorAuth', false);
-    const cachedAccountID = lodashGet(props, 'credentials.accountID', null);
+function ValidateLoginPage({account, credentials, route, session}: ValidateLoginPageProps<ValidateLoginPageOnyxProps>) {
+    const login = credentials?.login;
+    const autoAuthState = session?.autoAuthState ?? CONST.AUTO_AUTH_STATE.NOT_STARTED;
+    const accountID = Number(route?.params.accountID) ?? -1;
```

Comment:
Any reason for converting this to number? (was string before)
Number() is not nullish so there's no case of fallback to -1

Reply:
@situchan we use accountID as a number on the app but we get it as a string from the query parameter on the web app. 

You're right, Number is not nullish so I'll change this to `const accountID = Number(route?.params.accountID ?? -1);`. What do you think?

Reply:
That's fine. But I am not sure there's case of accountId not exist in param. If so, it should behave weirdly in UI.

Reply:
@situchan After taking a closer look into the code I finally realized that my logic didn't make sense 😅 accountID is no query param, and it will always be defined. In case the user tries to access the page, the app router will simply display a not found page.

So I removed all the default values and optional chaining.

Thanks again for your feedback @situchan  😄